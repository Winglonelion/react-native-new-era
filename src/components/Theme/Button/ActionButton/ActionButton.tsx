import React, { memo } from 'react';
import {
  TouchableWithoutFeedback,
  GestureResponderEvent,
  View,
  OpaqueColorValue,
} from 'react-native';
import ContentText from 'components/Base/Text/ContentText/ContentText';

interface PropTypes {
  text?: string;
  onPress?: (event: GestureResponderEvent) => void;
  disabled?: boolean;
  tint?: OpaqueColorValue | string;
}

const ActionButton: React.FC<PropTypes> = memo(
  ({ text = '', onPress, disabled = false, tint }) => {
    return (
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View>
          <ContentText color={tint}>{text.toUpperCase()}</ContentText>
        </View>
      </TouchableWithoutFeedback>
    );
  },
);
export default ActionButton;
