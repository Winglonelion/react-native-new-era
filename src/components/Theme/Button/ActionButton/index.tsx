export { default as default } from './ActionButton';
