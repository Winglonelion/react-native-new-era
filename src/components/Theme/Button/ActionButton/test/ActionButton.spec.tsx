import React from 'react';
import { shallow } from 'enzyme';
import ActionButton, { PropTypes } from '../ActionButton';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<ActionButton {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('ActionButton check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      text: 'button 1',
      disabled: false,
      onPress: jest.fn(),
      tint: 'blue',
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot with disabled', () => {
    const { wrapper } = setup({
      disabled: true,
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot without props', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
