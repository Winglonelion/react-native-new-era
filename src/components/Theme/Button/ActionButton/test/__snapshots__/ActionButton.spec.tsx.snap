// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton check snapshot should match snapshot with disabled 1`] = `
<TouchableWithoutFeedback
  disabled={true}
>
  <View>
    <Memo() />
  </View>
</TouchableWithoutFeedback>
`;

exports[`ActionButton check snapshot should match snapshot with props 1`] = `
<TouchableWithoutFeedback
  disabled={false}
  onPress={[MockFunction]}
>
  <View>
    <Memo()
      color="blue"
    >
      BUTTON 1
    </Memo()>
  </View>
</TouchableWithoutFeedback>
`;

exports[`ActionButton check snapshot should match snapshot without props 1`] = `
<TouchableWithoutFeedback
  disabled={false}
>
  <View>
    <Memo() />
  </View>
</TouchableWithoutFeedback>
`;
